api/data_explorer/controllers/facets_controller.py

```diff
@@ -20,32 +19,52 @@ def _get_bucket_interval(facet):
 
 
 def _process_extra_facets(extra_facets):
-    es = Elasticsearch(current_app.config['ELASTICSEARCH_URL'])
-
-    facets = OrderedDict()
-
     if not extra_facets:
         current_app.config['EXTRA_FACET_INFO'] = {}
         return
 
-    for elasticsearch_field_name in extra_facets:
-        if not elasticsearch_field_name:
+    es = Elasticsearch(current_app.config['ELASTICSEARCH_URL'])
+    facets = OrderedDict()
+    mapping = es.indices.get_mapping(index=current_app.config['INDEX_NAME'])
+
+    for es_base_field_name in extra_facets:
+        if not es_base_field_name:
             continue
 
-        arr = elasticsearch_field_name.split('.')
-        ui_facet_name = arr[-1]
-        field_type = elasticsearch_util.get_field_type(
-            es, elasticsearch_field_name)
-        facets[elasticsearch_field_name] = {
-            'ui_facet_name': ui_facet_name,
-            'type': field_type
-        }
-        facets[elasticsearch_field_name][
-            'description'] = elasticsearch_util.get_field_description(
-                es, elasticsearch_field_name)
-        facets[elasticsearch_field_name][
-            'es_facet'] = elasticsearch_util.get_elasticsearch_facet(
-                es, elasticsearch_field_name, field_type)
+        if elasticsearch_util.is_time_series(es, es_base_field_name):
```

Comment:
I would write this as 

is_time_series = elasticsearch_util.is_time_series(es, es_base_field_name)
if is_time_series:
    ....